app/src/main/java/org/fossasia/pslab/activity/MultimeterActivity.java

```diff
@@ -149,7 +150,26 @@ public void onClick(View view) {
             public void onClick(View view) {
                 if (scienceLab.isConnected()) {
                     Double capacitance = scienceLab.getCapacitance();
-                    saveAndSetData(String.valueOf(capacitance), getString(R.string.capacitance_unit));
+                    DecimalFormat capacitanceFormat = new DecimalFormat("#.##");
+                    String Capacitance;
+                    String capacitanceUnit;
+                    if (capacitance < 1e-9) {
+                        Capacitance = capacitanceFormat.format((capacitance / 1e-12));
+                        capacitanceUnit = "pF";
+                    } else if (capacitance < 1e-6) {
+                        Capacitance = capacitanceFormat.format((capacitance / 1e-9));
+                        capacitanceUnit = "nF";
+                    } else if (capacitance < 1e-3) {
+                        Capacitance = capacitanceFormat.format((capacitance / 1e-6));
+                        capacitanceUnit = "\u00B5"+"F";
+                    } else if (capacitance < 1e-1) {
+                        Capacitance = capacitanceFormat.format((capacitance / 1e-3));
+                        capacitanceUnit = "mF";
+                    } else {
+                        Capacitance = capacitanceFormat.format(capacitance);
+                        capacitanceUnit = getString(R.string.capacitance_unit);
+                    }
+                    saveAndSetData(Capacitance, capacitanceUnit);
```

Comment:
@yatri1609 These were the changes already done in my PR, please reset your git and then force push you commit.
Try `git reset --soft upstream/development` and then force push your commit.